streams/src/main/java/org/apache/kafka/streams/state/internals/ThreadCache.java

```diff
@@ -81,9 +85,13 @@ public synchronized void resize(final long newCacheSizeBytes) {
                 return;
             }
             final CircularIterator<NamedCache> circularIterator = new CircularIterator<>(caches.values());
-            while (sizeBytes() > maxCacheSizeBytes) {
+            while (sizeInBytes.get() > maxCacheSizeBytes) {
                 final NamedCache cache = circularIterator.next();
-                cache.evict();
+                synchronized (cache) {
```

Comment:
don't worry about this now as I want to merge your PR today, but WDYT about extracting this pattern of `lock - action - update_size` into a helper method where you can just pass in the action eg as `() -> flush()`? 

Not in a style/simplify the code way, but having a method like `#updateCacheAndSize` or similar might help future devs/us remember to follow this pattern instead of just directly invoking `flush`/`evict`/etc -- thoughts?

Reply:
If that makes sense to you feel free to do in a followup PR or just file a ticket, there are plenty of people looking to dip their toes into AK with simple patches like that.

Not at all married to the idea, though, so also feel free to ignore 🙂